modules/service/src/main/resources/graphql/itc.graphql

```diff
@@ -4576,39 +4576,53 @@ scalar NonNegBigDecimal
 """An `Long` in the range from 1 to 9223372036854775807"""
 scalar PosLong
 
-##############
-# ITC schema
-##############
+# Equivalent time amount in several unit options (e.g., 120 seconds or 2 minutes)
+type TimeSpan {
+  # TimeSpan in µs
+  microseconds: Long!
 
-# ITC error result
-type ItcError implements ItcResult {
-  # Message
-  msg: String!
+  # TimeSpan in ms
+  milliseconds: BigDecimal!
 
-  # Result type
-  resultType: String!
-}
+  # TimeSpan in seconds
+  seconds: BigDecimal!
+
+  # TimeSpan in minutes
+  minutes: BigDecimal!
 
-# Possible itc result
-interface ItcResult {
-  # Result type
-  resultType: String!
+  # TimeSpan in hours
+  hours: BigDecimal!
+
+  # TimeSpan as an ISO-8601 string
+  iso: String!
 }
 
-type ItcSuccess implements ItcResult {
-  # Exposure time duration
-  exposureTime: Duration!
+##############
+# ITC schema
+##############
 
-  # Exposures
-  exposures: Int!
+type Query {
+  # Given an instrument and a mode, return an the integration time and the estimated S/N
+  spectroscopyIntegrationTime(
+    input: SpectroscopyModeInput!
+  ): ExposureTimeCalculationResult!
 
-  # Signal/Noise ratio
-  signalToNoise: SignalToNoise!
+  spectroscopySignalToNoiseBeta(
+    input: SpectroscopySNInput!
```

Comment:
A tiny, tiny, wee little detail but it might be good if it were `SignalToNoise` everywhere in the schema (it's only here that it is `SN`)?  We could also make the inputs match the query fields?

```
spectroscopyIntegrationTime(
  input: SpectroscopyIntegrationTimeInput!
): ExposureTimeCalculationResult!
```
 and
```
spectroscopySignalToNoise(
  input: SpectroscopySignalToNoiseInput!
): SignalToNoiseResult!
```
and
```
spectroscopyGraph(
  input: SpectroscopyGraphInput!
): SpectroscopyGraphResult!
```

When you have `imagingIntegrationTime` will the result also be an `ExposureTimeCalculationResult` (i.e., not specific to spectroscopy vs. imaging)?

Reply:
Good idea, I think the return for imaging will be the same for imaging and spectroscopy, the return type takes a mode which is a union of `ImagindMode | SpectroscopyMode`